ts/features/bonus/bpd/screens/onboarding/BpdInformationScreen.tsx

```diff
@@ -17,21 +21,35 @@ export type Props = ReturnType<typeof mapDispatchToProps> &
  * This Screen shows all the information about the bpd program, with the rules and t&c.
  */
 
-const BpdInformationScreen: React.FunctionComponent<Props> = (props: Props) => (
-  <>
-    {props.bonus ? (
-      <BonusInformationComponent
-        bonus={props.bonus}
-        onConfirm={props.userActivateBpd}
-        onCancel={props.onCancel}
-      />
-    ) : null}
-  </>
-);
+const BpdInformationScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const onConfirm = () =>
+    getValue(props.bpdActiveBonus) === true
```

Comment:
The bpd activation is prefetched in the wallet, but we should also handle the case if it is still `RemoteLoading` or `RemoteError`, we no longer have the guarantee that the data will be loaded when we arrive on this page, having removed the check that guaranteed it in the saga.

One solution could be restore the saga part:
```typescript
  // read if the bpd is active for the user
  const isBpdActive: SagaCallReturnType<typeof isBpdEnabled> = yield call(
    isBpdEnabled
  );
```

in order to have the error /retry logic, and after that check we can continue to the InformationScreen with the safeness to have the data loaded.
Otherwise we have to handel the loading and the error inside the `BpdInformationScreen`.
We could see together tomorrow!